js/modules/views/initialization.js

```diff
@@ -0,0 +1,33 @@
+/* eslint-env browser */
+
+/* global i18n: false */
+
+
+const OPTIMIZATION_MESSAGE_DISPLAY_THRESHOLD = 2000; // milliseconds
```

Comment:
What lead you to the 2s timeframe here? Off the top of my head, I would have guessed something like 500ms.

Reply:
The thinking was that if startup process takes a 2 seconds or less, we would just show **Loading...** (from static page and later `AppLoadingScreen`). Only when we go to a longer time span (>= 2s) we would switch over to **Optimizing application…**. I’d like to address cases like Josh pointed out where the migration lasts on the order of minutes. As a compromise, I reduced it to 1 second:

- 0.5 seconds: http://g.recordit.co/DUYgige0RR.gif
- 1 second: http://recordit.co/pqtwqPAgxI (less distracting in my opinion)